pandas/core/tools/times.py

```diff
@@ -80,17 +80,20 @@ def _convert_listlike(arg, format):
             format_found = False
             for element in arg:
                 time_object = None
-                for time_format in formats:
-                    try:
-                        time_object = datetime.strptime(element, time_format).time()
-                        if not format_found:
-                            # Put the found format in front
-                            fmt = formats.pop(formats.index(time_format))
-                            formats.insert(0, fmt)
-                            format_found = True
-                        break
-                    except (ValueError, TypeError):
-                        continue
+                try:
```

Comment:
Why is this reordering necessary? Do you want to ensure that fromisoformat is called first?

Reply:
Yeah I was thinking it may be more common(?) for `fromisoformat` to successfully parse a time as opposed to checking a fixed list of 8 arbitrary formats. Of course I could try `fromisoformat` last too.

What do you think?

Reply:
No this is fine, just wanted to understand the reasoning behind the change